lib/rules/unused-module.js

```diff
@@ -0,0 +1,12 @@
+const {imported} = require('./dependency-graph')
+
+module.exports = function(context) {
+  return {
+    Program(node) {
+      const {filenames} = imported()
+      if (!filenames.has(context.getFilename())) {
+        context.report(node, 'Module was not imported by any files.')
```

Comment:
The last gotcha is we need a way to ignore "module entry" points. They won't be imported by any other file. In github's usecase, that's our `*-bootstrap.js` files. They do all the importing but we're going to link to them with a script tag, not another `import`.